### ISSUE TITLE
Add a possibility to define e-mail account/alias/group for Cloud SQL maintenance notification

### ISSUE THREAD

**Issue by zculek-fb** [hashicorp/terraform-provider-google#15375]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

At the moment, it is not possible to define custom e-mail address/alias (e.g. team e-mail alias) for Cloud SQL instance maintenance notifications - https://cloud.google.com/sql/docs/mysql/maintenance#:~:text=Notifications%20are%20sent%20to%20the%20email%20address%20associated%20with%20your%20Google%20account.%20It%27s%20not%20possible%20to%20configure%20a%20custom%20email%20alias%20(for%20instance%2C%20a%20team%20email%20alias). :

`[Notifications](https://cloud.google.com/error-reporting/docs/notifications#email) are sent to the email address associated with your Google account. It's not possible to configure a custom email alias (for instance, a team email alias).`

The way how it works now is that you can only opt-in for maintenance notifications which are sent to your Google account, and you receive notifications for all Cloud SQL instance, meaning that it's also not possible to enable notifications per instance, but only globally, for all instances. This is OK for sysadmins, SREs, IT gurus and alikes, but for companies that have different dev teams using their own Cloud SQL instance, it's just horrible that every person from the team has to enable maintenance notifications on their own, and then they receive notifications for all the company's Cloud SQL instances, even though they're only interested in one, instead of having the ability to define a team e-mail alias/group for every instance, which would then deliver maintenance notifications for all of the team members and only for their Cloud SQL instance.

### New or Affected Resource(s)

* google_sql_database_instance

### Potential Terraform Configuration

Defining maintenance notification e-mail address/alias could maybe be added to e.g. `settings.maintenance_window.` Perhaps it's not the best place to be added since option is not related with the maintenance window definition, but `settings.maintenance_window` also has `update_track` option which is not directly related to the maint window but rather to the maint itself. So, something like:
```tf
    maintenance_window {
      day          = maintenance_window_day
      hour         = maintenance_window_hour
      update_track = maintenance_window_update_track
      notification_email = maintenance_window_notification_email_address
    }
```

**Comment by rileykarson** [comment]
Note: This is a restriction in the API and not specific to Terraform. We can close this once the feedback reaches the SQL team.

**Comment by zculek-fb** [comment]
@rileykarson Thanks for the note! Is there anything else needed for this to reach the right crew?

**Comment by rileykarson** [comment]
Nope, this has now beed filed against them (as of a few days ago)
